metabase/metabase
# Feature | Collection Metadata

It's really hard to know in Metabase right now who made or has been recently working with a given dashboard, question, or collection

It's a feature branch, so it contains a set of smaller PRs:

- [x] Switch the collection page to table-based layout, displaying last edit info (#16357)
- [x] Update collection items drag-n-drop (#16360)
- [x] Add collection items sorting with the new backend API #16275  (#16385)
- [x] Fix UI flashes when sorting or switching collection items pages (#16511)
- [x] Display last edit info in Question and Dashboard headers (#16354)
- [x] Add tooltip displaying last edit time like "June 3, 12:10 PM" (#16492)
- [x] Add E2E tests for last edit info in Question and Dashboard headers (#16538)
- [x] Allow sorting by `last_edited_by` (#16388) and followup fixes (#16520, #16536)

Closes #15773, closes #6750, closes #13435

Notes:

- "Last edited at" and "Last edited by" columns are not displayed for Pulses. We retrieve this info from the most recent revision, and Pulses don't have them
- pinned items' `collection_position` field won't affect their position in the list anymore
- items filtering by their type is replaced with sorting (questions first, dashboards first)

## To Verify

1. Open a collection. The more different items it contains, the better
2. Ensure both pinned and not pinned tables are sorted alphabetically by name as a default
3. Click the "Name" column, the list has to be sorted in reversed alphabetical order
4. Click the "Type" column; the items have to be sorted in Dashboards-Pulses-Questions order
5. Click the "Type" column one more time; the items have to be sorted in Questions-Pulses-Dashboards order
6. Click the "Last edited at" column; the items have to be sorted by that column now
7. Hover any "Last edited at" cell, there should be a tooltip displaying the exact time of an edit 
7. Try applying different sortings to pinned and not pinned tables. They have to be separately sortable and not affect one another
8. Try editing dashboards and questions, the "last edited at" and "last edited by" should update

## Demo

### Before

<img width="1095" alt="List (before)" src="https://user-images.githubusercontent.com/17258145/120652475-7c9a1080-c488-11eb-8ada-254a20c7f049.png">

### After

https://user-images.githubusercontent.com/17258145/121322453-a71c1b80-c917-11eb-823e-3db8afe23852.mov
